src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -233,876 +233,953 @@ void GetPurchasedAir(EnergyPlusData &state)
-            PurchAir(PurchAirNum).HtRecTotHeatEnergy = 0.0;
-            PurchAir(PurchAirNum).HtRecSenCoolEnergy = 0.0;
-            PurchAir(PurchAirNum).HtRecLatCoolEnergy = 0.0;
-            PurchAir(PurchAirNum).HtRecTotCoolEnergy = 0.0;
-            PurchAir(PurchAirNum).SenHeatRate = 0.0;
-            PurchAir(PurchAirNum).LatHeatRate = 0.0;
-            PurchAir(PurchAirNum).TotHeatRate = 0.0;
-            PurchAir(PurchAirNum).SenCoolRate = 0.0;
-            PurchAir(PurchAirNum).LatCoolRate = 0.0;
-            PurchAir(PurchAirNum).TotCoolRate = 0.0;
-            PurchAir(PurchAirNum).ZoneSenHeatRate = 0.0;
-            PurchAir(PurchAirNum).ZoneLatHeatRate = 0.0;
-            PurchAir(PurchAirNum).ZoneTotHeatRate = 0.0;
-            PurchAir(PurchAirNum).ZoneSenCoolRate = 0.0;
-            PurchAir(PurchAirNum).ZoneLatCoolRate = 0.0;
-            PurchAir(PurchAirNum).ZoneTotCoolRate = 0.0;
-            PurchAir(PurchAirNum).OASenHeatRate = 0.0;
-            PurchAir(PurchAirNum).OALatHeatRate = 0.0;
-            PurchAir(PurchAirNum).OATotHeatRate = 0.0;
-            PurchAir(PurchAirNum).OASenCoolRate = 0.0;
-            PurchAir(PurchAirNum).OALatCoolRate = 0.0;
-            PurchAir(PurchAirNum).OATotCoolRate = 0.0;
-            PurchAir(PurchAirNum).HtRecSenHeatRate = 0.0;
-            PurchAir(PurchAirNum).HtRecLatHeatRate = 0.0;
-            PurchAir(PurchAirNum).HtRecTotHeatRate = 0.0;
-            PurchAir(PurchAirNum).HtRecSenCoolRate = 0.0;
-            PurchAir(PurchAirNum).HtRecLatCoolRate = 0.0;
-            PurchAir(PurchAirNum).HtRecTotCoolRate = 0.0;
-
-            PurchAir(PurchAirNum).OutdoorAirMassFlowRate = 0.0;
-            PurchAir(PurchAirNum).OutdoorAirVolFlowRateStdRho = 0.0;
-            PurchAir(PurchAirNum).SupplyAirMassFlowRate = 0.0;
-            PurchAir(PurchAirNum).SupplyAirVolFlowRateStdRho = 0.0;
+            // get optional heating fuel efficiency schedule name
+            cAlphaFieldName = "Heating Fuel Efficiency Schedule Name";
+            std::string const heatFuelEfficiencySchedName = s_ip->getAlphaFieldValue(fields, schemaProps, "heating_fuel_efficiency_schedule_name");
+            if (heatFuelEfficiencySchedName.empty()) {
+                PurchAir.heatFuelEffSched = Sched::GetScheduleAlwaysOn(state);
+            } else if ((PurchAir.heatFuelEffSched = Sched::GetSchedule(state, heatFuelEfficiencySchedName)) == nullptr) {
+                ShowSevereItemNotFound(state, eoh, cAlphaFieldName, heatFuelEfficiencySchedName);
+                ErrorsFound = true;
+            }
+            // get heating fuel type
+            cAlphaFieldName = "Heating Fuel Type";
+            fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "heating_fuel_type");
+            PurchAir.heatingFuelType = static_cast<Constant::eFuel>(getEnumValue(Constant::eFuelNamesUC, Util::makeUPPER(fieldValue)));
+            if (PurchAir.heatingFuelType == Constant::eFuel::Invalid) {
```

Comment:
`Invalid` is not possible. This code will never execute. Delete.

Reply:
Done.